script/libtest.sh

```diff
@@ -12,37 +12,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+set -e
+
 export TEST_MAX_WAIT_SEC=300
 
 ## k8s specific Helper functions
-k8s_wait_for_pod() {
-    namespace=${1:?}
-    labelSelector=${2:?}
-    condition=${3:?}
-    echo "Waiting for pod '${@:2}' to be ${condition} ... "
-    local -i cnt=${TEST_MAX_WAIT_SEC:?}
-
-    # Retries just in case it is not stable
-    local -i successCount=0
-    while [ "$successCount" -lt "3" ]; do
-        if kubectl get pod -a -n "$namespace" -l "$labelSelector" | grep -q "$condition"; then
-            ((successCount=successCount+1))
-        fi
-        ((cnt=cnt-1)) || return 1
-        sleep 1
-    done
-
-}
-k8s_wait_for_pod_ready() {
-    namespace=${1:?}
-    labelSelector=${2:?}
-    k8s_wait_for_pod $namespace $labelSelector Running
-}
-
-k8s_wait_for_pod_completed() {
+k8s_wait_for_deployment() {
     namespace=${1:?}
-    labelSelector=${2:?}
-    k8s_wait_for_pod $namespace $labelSelector Completed
+    deployment=${2:?}
+    echo "Waiting for deployment ${deployment} to be successfully rolled out"
```

Comment:
I see that compared to #833 you have decided not to check rollout multiple times. Any reason for that change? https://github.com/kubeapps/kubeapps/pull/833/files#diff-f92c3290968ffccd70b499c19fdeaba2R37

Reply:
I didn't really understood why retrying on a rollout, the command will either wait (and hang) until the rollout success or fail (and won't success even if you retry) so I am not sure why we need to retry. That may be useful in the case the deployment has not been created yet but that will never be the case here since the deployment is created with `helm install`.